app/src/main/java/com/teamagam/gimelgimel/app/map/actions/freedraw/FreeDrawer.java

```diff
@@ -0,0 +1,229 @@
+    mCommands = new Stack<>();
+    mEntityByIdMap = new HashMap<>();
+    mIsDrawingMode = true;
+    mIsEnabled = true;
+  }
+
+  public void start() {
+    sLogger.v("Started free drawing");
+    mGgMapView.getMapDragEventObservable()
+        .filter(mde -> mIsEnabled && mIsDrawingMode)
+        .doOnNext(this::log)
+        .filter(this::containsDrag)
+        .subscribe(this::drawDragEvent);
+  }
+
+  public void undo() {
+    if (!mCommands.isEmpty()) {
+      mCommands.pop().undo();
+    } else {
+      sLogger.v("No action to undo");
+    }
+  }
+
+  public void setColor(String color) {
+    mSymbol = createSymbol(color);
+  }
+
+  public void switchMode() {
+    toggle();
+    mGgMapView.setOnEntityClickedListener(mIsDrawingMode ? null : new EraserListener());
+  }
+
+  public boolean isInEraserMode() {
+    return !mIsDrawingMode;
+  }
+
+  public void enable() {
+    mIsEnabled = true;
+  }
+
+  public void disable() {
+    mIsEnabled = false;
+  }
+
+  private void log(MapDragEvent mde) {
+    sLogger.v("drag-event: "
+        + mde.getFrom().getLongitude()
+        + ","
+        + mde.getFrom().getLatitude()
+        + " - "
+        + mde.getTo().getLongitude()
+        + ","
+        + mde.getTo().getLatitude());
+  }
+
+  private boolean containsDrag(MapDragEvent mde) {
+    return !mde.getFrom().equals(mde.getTo());
+  }
+
+  private void drawDragEvent(MapDragEvent mde) {
+    if (isNewDragStream(mde)) {
+      resetStream(mde);
+    } else {
+      hideLastDrawnEntity();
+    }
+    mCurrentPoints.add(mde.getTo());
+    displayCurrent();
+  }
+
+  private boolean isNewDragStream(MapDragEvent mde) {
+    return mCurrentPoints == null || !isConnectedToCurrentStream(mde);
+  }
+
+  private boolean isConnectedToCurrentStream(MapDragEvent mde) {
+    PointGeometry lastTo = mCurrentPoints.get(mCurrentPoints.size() - 1);
+    PointGeometry currentFrom = mde.getFrom();
+    return abs(currentFrom.getLatitude() - lastTo.getLatitude()) <= mTolerance
+        && abs(currentFrom.getLongitude() - lastTo.getLongitude()) <= mTolerance;
+  }
+
+  private void resetStream(MapDragEvent mde) {
+    mCurrentPoints = new ArrayList<>();
+    mCurrentPoints.add(mde.getFrom());
+  }
+
+  private void hideLastDrawnEntity() {
+    mCommands.pop().undo();
+  }
+
+  private void displayCurrent() {
+    PolylineEntity p = mMapEntityFactory.createPolyline(mCurrentPoints, mSymbol);
+    addDisplayCommand(p);
+  }
+
+  private void addDisplayCommand(PolylineEntity p) {
+    DisplayCommand cmd = new DisplayCommand(p);
+    cmd.execute();
+    mCommands.push(cmd);
+  }
+
+  private PolylineSymbol createSymbol(String color) {
+    return new PolylineSymbol.PolylineSymbolBuilder().setBorderColor(color).build();
+  }
+
+  private void toggle() {
+    mIsDrawingMode = !mIsDrawingMode;
+  }
+
+  private interface Command {
+    void execute();
+
+    void undo();
+  }
+
+  private abstract class MapActionCommand implements Command {
```

Comment:
It seems like this isn't really a classic use of an inheritance as the inheriting classes don't get any "benefit" (/behavior) if they inherit from this base class unless they specifically call these newly introduced protected methods.
In this case, I'd rather use composition here (meaning, MapActionCommand will not implement command, but it'll be used to implement the command interface in the inheriting classes).
The main benefit is leaving the actual commands "free" to inherit, if needed, from a more meaningful base-class (one that will add some behavior to its descendants)